### PR TITLE
New Angular 2 Cheat Sheet

### PR DESCRIPTION
**Pull request by vikashvverma**
**What does your Instant Answer do?**

Displays a list of Angular 2 built-in template syntax, directives, components, decorators, dependency injection and their configuration in response to searches like `angular 2 cheatsheet`, `angular 2 cheat sheet`

**What is the data source for your Instant Answer? (Provide a link if possible)**

[ https://angular.io/docs/ts/latest/guide/cheatsheet.html](https://angular.io/docs/ts/latest/guide/cheatsheet.html)

**Why did you choose this data source?**

This is official source for Angular 2.

**Are there any other alternative (better) data sources?**

Since the source is the official Angular 2, I don't think there is any other better source.

**Which communities will this Instant Answer be especially useful for? (gamer, book lovers, etc)**

Programmers

**Is this Instant Answer connected to a DuckDuckHack Instant Answer idea?**

No

**Are you having any problems? Do you need our help with anything?**

I don't have any problems as of now but I would like to setup everything locally for better productivity.

**Where did you hear about DuckDuckHack? (For first time contributors)**

I heard about this at [ http://www.meetup.com/Quack-Hack-Bangalore/](http://www.meetup.com/Quack-Hack-Bangalore/)

**What does the Instant Answer look like? (Provide a screenshot for new or updated Instant Answers)**

screen shot 2016-01-31 at 6 26 15 pm

IA Page: [ https://duck.co/ia/view/angular_2_cheat_sheet](https://duck.co/ia/view/angular_2_cheat_sheet)

### CODE REVIEW

**Comment by daxtheduck** [comment]
@vikashvverma Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

While the team and the community review your submission, we'd like to make sure you look good! If you'd like a free DuckDuckGo tshirt, please send over the following info to merch@duckduckgo.com, and be sure to put "Dev Merch" in the subject line:

Name:
Street Address:
City:
State / Province:
Country:
Phone (Non-US Orders only): 
T-shirt size:
GitHub username:
Link to this Pull Request:

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please send an email to <quackslack@duckduckgo.com>.

>More Info: https://duck.co/duckduckhack/submission_and_review

Thanks!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by jophab** [comment]
@vikashvverma Welcome to Duck Duck Hack :smile: 
Good job :+1:

File: share/goodie/cheat_sheets/json/angular2.json [hunk 1]
```diff
@@ -0,0 +1,97 @@
+{
+    "id": "angular2_cheat_sheet",
+    "name": "Angular2",
+    "description": "Angular 2 for TypeScript Cheat Sheet ",
+    "metadata": {
+        "sourceName": "Angular",
+        "sourceUrl": "https://angular.io/docs/ts/latest/guide/cheatsheet.html"
+    },
+    "aliases": ["angular2", "angularjs 2", "angular 2"],
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/angular2.json]: @vikashvverma first alias is same as your file name. so no need of it.

File: share/goodie/cheat_sheets/json/angular2.json [hunk 2]
```diff
@@ -0,0 +1,97 @@
+{
+    "id": "angular2_cheat_sheet",
+    "name": "Angular2",
+    "description": "Angular 2 for TypeScript Cheat Sheet ",
```
Comment (jophab) [inline share/goodie/cheat_sheets/json/angular2.json]: @vikashvverma give some decription about the contetns of the cheat sheet here.
Shorter is preferable. Or else we can avoid it.

Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/angular2.json]: I will make the changes.

**Comment by vikashvverma** [comment]
@moollaza Can you review and merge this pull request?

**Comment by jophab** [comment]
@vikashvverma Lets remove periods from end of val fields and description.
Also,If possible shall we try to make the val fields a little shorter?

**Comment by vikashvverma** [comment]
@moollaza Can you review this?

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Angular 2 Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/angular_2_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/angular2.json [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+            "val": "Binds the presence of the CSS class extra-sparkle on the element to the truthiness of the expression isDelightful"
+        }, {
+            "key": "<div \\[style.width.px\\]=\"mySize\">",
+            "val": "Binds style property width to the result of expression mySize in pixels. Units are optional"
+        }, {
+            "key": "<button (click)=\"readRainbow($event)\">",
+            "val": "Calls method readRainbow when a click event is triggered on this button element (or its children) and passes in the event object"
+        }, {
+            "key": "<p>Hello \\{\\{greeting\\}\\}</p>",
+            "val": "Binds text content to an interpolated string, e.g. \"नमस्ते\""
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/angular2.json]: Please use English for the example -- everything else here is English :+1:

File: share/goodie/cheat_sheets/json/angular2.json [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        }, {
+            "key": "<my-cmp \\[(title)\\]=\"name\">",
+            "val": "Sets up two-way data binding. Equivalent to: <my-cmp \\[title\\]=\"name\" (titleChange)=\"name=$event\">"
+        }, {
+            "key": "<p *myUnless=\"myExpression\">...</p>",
+            "val": "The * symbol means that the current element will be turned into an embedded template. Equivalent to: <template \\[myUnless\\]=\"myExpression\"><p>...</p></template>"
```
Comment (moollaza) [inline share/goodie/cheat_sheets/json/angular2.json]: Let's drop the `Equivalent to ...` it looks odd because it's not in a code block

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Angular 2 Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/angular_2_cheat_sheet_mobile.png)

File: share/goodie/cheat_sheets/json/angular2.json (hunk 4 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/angular2.json]: Done!

File: share/goodie/cheat_sheets/json/angular2.json (hunk 3 above)
Comment (vikashvverma) [inline share/goodie/cheat_sheets/json/angular2.json]: Done!

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Angular 2 Cheat Sheet)
                
                [Mobile screenshot](https://ia-screenshots.s3.amazonaws.com/angular_2_cheat_sheet_mobile.png)

**Comment by moollaza** [comment]
@vikashvverma This LGTM :+1:

Thanks again for taking the time to contribute! We really appreciate it :smiley:

**Comment by vikashvverma** [comment]
@moollaza This was my first PR to DuckDuckGo :smile:

**Pull request merged**